File: backend/src/main/java/com/optimiza/clickbarber/repository/ClienteRepository.java
package com.optimiza.clickbarber.repository;

import com.optimiza.clickbarber.model.Cliente;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface ClienteRepository extends JpaRepository<Cliente, Long> {

    Optional<Cliente> findByUsuarioId(Long usuarioId);

}

File: backend/src/main/java/com/optimiza/clickbarber/repository/ClienteRepository.kt
package com.optimiza.clickbarber.repository

import com.optimiza.clickbarber.model.Cliente
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ClienteRepository : JpaRepository<Cliente, Long> {
    fun findByUsuarioId(usuarioId: Long): Optional<Cliente>

    @Query("""
        SELECT c FROM Cliente c
        INNER JOIN Barbearia b ON b.idExterno = :idExternoBarbearia
        INNER JOIN Agendamento a ON a.cliente = c
        INNER JOIN Usuario u ON u.id = c.usuario.id
    """)
    fun findByIdExternoBarbearia(idExternoBarbearia: UUID): List<Cliente>
}